src/v/kafka/server/handlers/topics/topic_utils.cc

```diff
@@ -47,14 +48,16 @@ ss::future<std::vector<model::node_id>> wait_for_leaders(
             // topic already deleted
             continue;
         }
-        // for each partition ask for leader
-        for (auto& pmd : md->get().get_assignments()) {
-            futures.push_back(md_cache.get_leader(
-              model::ntp(r.tp_ns.ns, r.tp_ns.tp, pmd.id), timeout));
-        }
+        co_await ss::max_concurrent_for_each(
+          md->get().get_assignments(),
+          64,
+          [&r, &md_cache, timeout](const auto& p_as) {
+              return md_cache
+                .get_leader(
+                  model::ntp(r.tp_ns.ns, r.tp_ns.tp, p_as.id), timeout)
```

Comment:
Nit: what is called "timeout" here is actually a "deadline" (this made me to initially incorrectly assume that we can now get stuck here for `timeout * number_of_partitions/64` instead of just `timeout`)